engines/ic3ia.cpp

```diff
@@ -134,10 +134,20 @@ void IC3IA::initialize()
 {
   super::initialize();
 
+  Sort boolsort = solver_->make_sort(BOOL);
+  // add predicates automatically added by ia_
+  // to our predset_
+  // needed to prevent adding duplicate predicates later
+  for (const auto & p : ia_.predicates()) {
+    // ia_ automatically includes all boolean variables
+    assert(p->is_symbolic_const() && p->get_sort() == boolsort);
+    predset_.insert(p);
+  }
+
   // add all the predicates from init and property to the abstraction
   // NOTE: abstract is called automatically in IC3Base initialize
   UnorderedTermSet preds;
-  get_predicates(solver_, ts_.init(), preds, false, false, true);
+  get_predicates(solver_, conc_ts_.init(), preds, false, false, true);
```

Comment:
essentially ts_.init() and conc_ts_.init() are the same. It is just more clearer to use conc_ts_.
RIght?

Reply:
Yeah that's right, I actually only changed it because I got confused about it myself :P But yes they should be the same.